webui/react/src/components/DeterminedAuth.test.tsx

```diff
@@ -0,0 +1,86 @@
+import { render, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import UIProvider, { DefaultTheme } from 'hew/Theme';
+import { useInitApi } from 'hew/Toast';
+import { useState } from 'react';
+
+import { login } from 'services/api';
+
+import DeterminedAuth, {
+  PASSWORD_ID,
+  SUBMIT_ID,
+  USERNAME_ID,
+  WEAK_PASSWORD_SUBJECT,
+} from './DeterminedAuth';
+import { ThemeProvider } from './ThemeProvider';
+
+const USERNAME = 'username';
+const WEAK_PASSWORD = 'password';
+const STRONG_PASSWORD = 'dO2Ccf5ceozXJFuMhjEx32UgtR4X4yZTm1Vv';
+
+vi.mock('services/api', () => ({
+  login: vi.fn().mockResolvedValue({ token: '', user: {} }),
+}));
+
+const Container = (): JSX.Element => {
+  const [canceler] = useState(new AbortController());
+  useInitApi();
+  return <DeterminedAuth canceler={canceler} />;
+};
+
+const setup = () => {
+  render(
+    <UIProvider theme={DefaultTheme.Light}>
+      <ThemeProvider>
+        <Container />
+      </ThemeProvider>
+    </UIProvider>,
+  );
+};
+
+const user = userEvent.setup();
+describe('DeterminedAuth', () => {
+  it('logs in with strong password', async () => {
+    setup();
+
+    await user.type(screen.getByTestId(USERNAME_ID), USERNAME);
+    await user.type(screen.getByTestId(PASSWORD_ID), STRONG_PASSWORD);
+    user.click(screen.getByTestId(SUBMIT_ID));
+
+    await waitFor(() => {
+      // onFinish begins:
+      expect(screen.getByRole('button')).toBeDisabled();
+    });
+    await waitFor(() => {
+      // onFinish ends:
+      expect(screen.getByRole('button')).not.toBeDisabled();
```

Comment:
i dont think these assertions are working. even if i add/remove `.not` test cases pass

Reply:
it's expected that removing `not` from line 56 would not change pass/fail. these `waitFor`s are to make sure that the `onFinish` function has completed before checking the `not` on line 59, to avoid false positive there.

Reply:
same thing happens in line 51 too. adding `.not` doesnt fail the test case

Reply:
that's expected. if they're both have `not` or both don't have `not`, they no longer test that `onFinish` has started and completed. the point is not to test whether they pass or fail, but rather that they both must pass in order for the assertion on line 59 to be valid and not a false positive.

Reply:
open to a better way of preventing false positive on line 59 but that's the reasoning.

Reply:
got it